src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py

```diff
@@ -256,6 +256,51 @@ def test_stream_failed_run_logs(self, remote_client, pf, capfd):
         # error info will store in run dict
         assert "error" in run._to_dict()
 
+    def test_archive_and_restore_run(self, remote_client):
+        from promptflow._sdk._constants import RunHistoryKeys
+
+        run_meta_data = RunHistoryKeys.RunMetaData
+        hidden = RunHistoryKeys.HIDDEN
+
+        run_id = "4cf2d5e9-c78f-4ab8-a3ee-57675f92fb74"
+
+        # test archive
+        remote_client.runs.archive(run=run_id)
+        run_data = remote_client.runs._get_run_from_run_history(run_id, original_form=True)[run_meta_data]
+        assert run_data[hidden] is True
+
+        # test restore
+        remote_client.runs.restore(run=run_id)
+        run_data = remote_client.runs._get_run_from_run_history(run_id, original_form=True)[run_meta_data]
+        assert run_data[hidden] is False
+
+    def test_update_run(self, remote_client):
+        run_id = "4cf2d5e9-c78f-4ab8-a3ee-57675f92fb74"
```

Comment:
create your own run id randomly?
or create a flow run everytime and get the run id?

Reply:
also negative cases?

Reply:
As you can see there are some other tests also using existing run id to test the functionalities. The original purpose was to not rely on `pf run create` since run creation api was not stable atm. 

Now the api is much more stable so I will switch to creating new run for test. I have created a task to do so, so may not finish it in this pr

Reply:
negative case is added in line 295, for invalid input this update method won't fail just log warings.